### ISSUE TITLE
Add ability to add connector ports via JVM

### ISSUE THREAD

**Issue by ismagilov** [svenkubiak/mangooio#271]
I tried to deploy the simplest application, generated from Maven archetype, to Heroku. Heroku demands to set a listening port from the command line. You need to write something, like: `java -jar app.jar -Dserver.port=$PORT`, where `$PORT` is substituted with Heroku`s port value.

According with paragraph:
> If mangoo I/O can find a environment specific value (e.g. dev.application.port), the value will overwrite the default value (e.g. default.application.port).

I have tried to overwrite a HTTP port, but it didn\`t work (see the output line: `HTTP connector listening ...:*** 8080 ***`).

```
ilya$ java -jar target/mangooioapp.jar -Dprod.connector.http.port=8181

14.03.2017 18:23:09,911 [main] INFO  org.ehcache.core.EhcacheManager - Cache 'mangooio-application' created in EhcacheManager.
14.03.2017 18:23:09,917 [main] INFO  org.ehcache.core.EhcacheManager - Cache 'mangooio-auth' created in EhcacheManager.
14.03.2017 18:23:09,921 [main] INFO  org.ehcache.core.EhcacheManager - Cache 'mangooio-request' created in EhcacheManager.
14.03.2017 18:23:09,924 [main] INFO  org.ehcache.core.EhcacheManager - Cache 'mangooio-sse' created in EhcacheManager.
14.03.2017 18:23:09,927 [main] INFO  org.ehcache.core.EhcacheManager - Cache 'mangooio-wss' created in EhcacheManager.
14.03.2017 18:23:09,983 [main] INFO  io.mangoo.providers.TemplateEngineProvider - Using class io.mangoo.templating.freemarker.TemplateEngineFreemarker as implementation for Template Engine
14.03.2017 18:23:10,241 [main] INFO  org.xnio - XNIO version 3.3.6.Final
14.03.2017 18:23:10,250 [main] INFO  org.xnio.nio - XNIO NIO Implementation Version 3.3.6.Final
14.03.2017 18:23:10,313 [main] INFO  io.mangoo.core.Bootstrap - 
                                                ___     __  ___  
 _ __ ___    __ _  _ __    __ _   ___    ___   |_ _|   / / / _ \ 
| '_ ` _ \  / _` || '_ \  / _` | / _ \  / _ \   | |   / / | | | |
| | | | | || (_| || | | || (_| || (_) || (_) |  | |  / /  | |_| |
|_| |_| |_| \__,_||_| |_| \__, | \___/  \___/  |___|/_/    \___/ 
                          |___/                                  


https://mangoo.io | @mangoo_io | 4.2.0

14.03.2017 18:23:10,314 [main] INFO  io.mangoo.core.Bootstrap - HTTP connector listening @localhost:8080
14.03.2017 18:23:10,316 [main] INFO  io.mangoo.core.Bootstrap - mangoo I/O application started in 1141 ms in prod mode. Enjoy.
```

**Comment by MarkVink** [comment]
I do not know if you can change the server port using VM arguments, but you'll be able to point to your production configuration file.

I'm running MangooIO applications on my server using supervisor and the following command; 
`java -jar -Dapplication.config=application.yaml ***.jar`

**Comment by ismagilov** [comment]
Not sure that this will help much in Heroku env. Port is known only in Heroku's runtime.

As an illustration, you can see an article describing how to run spring-boot app: https://devcenter.heroku.com/articles/deploying-spring-boot-apps-to-heroku. 

Command line used by Heruko looks like: `java $JAVA_OPTS -Dserver.port=$PORT ...`. So Heroku`s runtime takes this line. Substitutes $PORT with its value. And launches your app, forcing it to use externally set port value.

**Comment by svenkubiak** [comment]
The paragraph you qouted references to the application.yaml and its handling of different modes. 

https://docs.mangoo.io/modes.html
https://docs.mangoo.io/default-values.html

Passing variables to the VM is a different topic. At the moment it is only possible to add the mode, config path and log file path to the jvm.

```
-Dapplication.mode=foo
-Dapplication.config=/path/to/conf/application.yaml
-Dapplication.log=/path/to/conf/log4j2.yaml
```

Passing a connector port would probably a good enhancement. I'll add this to the 4.3.0 release.

**Comment by ismagilov** [comment]
Don\`t you think that it is good idea to let any application config values being overwritten from the command line? As far as I can see, *spring-boot* allows it. But as for [ninjaframework](http://www.ninjaframework.org/documentation/configuration_and_modes.html), looks like they limit the set of config values which can be overwritten from CLI.

And is there any workaround for this problem which can be used in 4.2.0 version?

**Comment by svenkubiak** [comment]
The purpose of the current approach is to change resource files without having to re-deploy - with application.mode being the exception. Besides having boiler-plate code, I don't see any advantages in having the option to have every conf value configurable via command line. I would rather adapt to what is required - as for the port in order to get it running on heroku.

You could bind the config class to your own implemntation and hook into the application lifecycle before undertow is started. But that is probably very hackish.

**Comment by svenkubiak** [comment]
I just pushed the changes to the 4.3.0 feature branch. Feel free to give it a try. New jvm attributes are

```
http.host
http.port
ajp.host
ajp.port
```

**Comment by ismagilov** [comment]
Many thanks for quick fix! I had spent more time then expected because I had skipped an option `-Dhttp.host=0.0.0.0`. By default mangoo opens port on `localhost`, but Heroku requires `0.0.0.0`.

Now it is working:
> 2017-03-21T06:29:46.640778+00:00 app[web.1]: https://mangoo.io | @mangoo_io | 4.3.0-Beta2
2017-03-21T06:29:46.640779+00:00 app[web.1]: 
2017-03-21T06:29:46.640927+00:00 app[web.1]: 21.03.2017 06:29:46,640 [main] INFO  io.mangoo.core.Bootstrap - HTTP connector listening @0.0.0.0:3547
2017-03-21T06:29:46.654609+00:00 app[web.1]: 21.03.2017 06:29:46,654 [main] INFO  io.mangoo.core.Bootstrap - mangoo I/O application started in 2334 ms in prod mode. Enjoy.
